go-controller/pkg/util/net_linux.go

```diff
@@ -478,7 +479,7 @@ func GetNetworkInterfaceIPs(iface string) ([]*net.IPNet, error) {
 
 	var ips []*net.IPNet
 	for _, addr := range addrs {
-		if addr.IP.IsLinkLocalUnicast() {
+		if addr.IP.IsLinkLocalUnicast() || isAddressReservedForInternalUse(addr.IP) {
```

Comment:
note IsLinkLocalUnicast() matches our use of 169.254 subnet :) but it does not match for ipv6!

Reply:
yeah, I wanted to make sure that these reserved IPs, also in IPv4, do not go unnoticed next time we change something around it.